kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -579,57 +581,71 @@ private boolean startConnectionMonitorTask() {
 
             // Change notification status to slow blinking when connection is expected to happen in the future
             this.cloudConnectionStatusService.updateStatus(this, CloudConnectionStatusEnum.SLOW_BLINKING);
-            // add a delay on the reconnect
-            int maxDelay = reconnectInterval / 5;
-            maxDelay = maxDelay > 0 ? maxDelay : 1;
-            int initialDelay = new Random().nextInt(maxDelay);
 
-            logger.info("Starting reconnect task with initial delay {}", initialDelay);
-            this.connectionMonitorFuture = this.connectionMonitorExecutor.scheduleAtFixedRate(new Runnable() {
+            this.connectionMonitorFuture = this.connectionMonitorExecutor.schedule(new Runnable() {
 
                 @Override
                 public void run() {
-                    String originalName = Thread.currentThread().getName();
-                    Thread.currentThread().setName("DataServiceImpl:ReconnectTask");
-                    boolean connected = false;
-                    try {
-                        if (DataServiceImpl.this.dbService == null) {
-                            logger.warn("H2DbService instance not attached, not connecting");
-                            return;
-                        }
-                        logger.info("Connecting...");
-                        if (DataServiceImpl.this.dataTransportService.isConnected()) {
-                            logger.info("Already connected. Reconnect task will be terminated.");
 
-                        } else {
-                            DataServiceImpl.this.dataTransportService.connect();
-                            logger.info("Connected. Reconnect task will be terminated.");
-                        }
-                        connected = true;
-                    } catch (KuraConnectException e) {
-                        logger.warn("Connect failed", e);
-
-                        if (DataServiceImpl.this.dataServiceOptions.isConnectionRecoveryEnabled()) {
-                            if (isAuthenticationException(e) || DataServiceImpl.this.connectionAttempts
-                                    .getAndIncrement() < DataServiceImpl.this.dataServiceOptions
-                                            .getRecoveryMaximumAllowedFailures()) {
-                                logger.info("Checkin done.");
-                                DataServiceImpl.this.watchdogService.checkin(DataServiceImpl.this);
+                    int attempts = 1;
+                    int backoffMaxTentatives = DataServiceImpl.this.dataServiceOptions.getBackoffMaxTentatives();
+
+                    while (!DataServiceImpl.this.dataTransportService.isConnected() || autoConnect) {
+
+                        int reconnectDelay = (new Random().nextInt((int) Math.pow(2, attempts)) + 1) * 1000;
```

Comment:
According to [this document](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/), the reconnect delay should be implemented as
`sleep = random_between(0, min(cap, base * 2 ** attempt))`
So, it is assumed that the base is 1 (this is fine), but the cap is missing.